### PR TITLE
Two ways of testing your API using LoadImpact v4

### PR DESCRIPTION
**Pull request by sniku**

### CODE REVIEW

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+This means that we need `50 VUs` to generate `200` requests every second.
+
+You can view a sample test run of this script [here](https://app.loadimpact.com/k6/anonymous/81e9e44f12cb49c88da707fbdc65ee4a)
+
+![k6 rps with VUs]({{ site.baseurl }}/assets/img/v4/how-to-tutorials/how-to-load-test-an-api/rps_result.png)
+
+
+## Executing tests and viewing results
+
+If your API is publicly accessible on the internet, you can run this script from LoadImpact cloud by running this command:
```
Comment (robingustafsson) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: "LoadImpact" => "Load Impact"

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+{% highlight console %}
+$ k6 cloud API_scenario_test.js
+{% endhighlight %}
+
+If your API is behind a firewall or running on your development machine, you can use local execution, running this command:
+{% highlight console %}
+$ k6 run -o API_scenario_test.js
```
Comment (robingustafsson) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: This is missing `cloud` after `-o`

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+{% endhighlight %}
+
+In both cases you will see a link to Load Impact where you can view your results in real time. It will look similar to this:
+
+![k6 output]({{ site.baseurl }}/assets/img/v4/how-to-tutorials/how-to-load-test-an-api/k6_cloud_output.png)
+
+
+
+Note, if you are just getting started or don't want to mess with terminal, you can also execute scripts directly from Load Impact's web editor, without the need of installing k6. Go to the
```
Comment (robingustafsson) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: Think there should a "a" or "the" before "terminal" here.

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+***
+
+Load testing is used to test scalability and performance of sites, apps and APIs;
+there are however some differences in how to prepare and configure your load test depending on your type of system and what answer you are looking to get from your testing.
+
+There are two typical approaches to testing APIs:
+ 1. You can test APIs in the same way you test websites, by simulating users (or VUs). You want to use this method if you are interested in how many users your API can serve simultaneously.
+ 2. You can test API for number or requests per second (or RPS). You want to use this method, if your are interested in how many requests your API can handle per second.
+
+If you are testing one endpoint at a time, the RPS method may be more appropriate. If you are testing a scenario with multiple endpoints, the VU method is probably more meaningful.
```
Comment (markjmeier) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: Might be clearer to describe it as "If you are a testing a real/logical user journey across multiple endpoints, the VU method is more meaningful"

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+You can view a sample test run of this script [here](https://app.loadimpact.com/k6/anonymous/4eacae1dfd234566865962ccdb350c4e)
+
+![k6 rps with VUs]({{ site.baseurl }}/assets/img/v4/how-to-tutorials/how-to-load-test-an-api/rps_vus.png)
+
+Note, in this test run, the Request Rate closely follows VUs, while response time is relatively flat.
+This means that the API under test is handling traffic well for specified number or users.
+
+# Testing API endpoints for RPS
+
+By default, k6 operates in the "VU mode", but you can easily limit how many RPS each VU executes, so it's easy to simulate RPS!
```
Comment (markjmeier) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: Suggestion: By default, k6 will operate in "VU mode". However, you can easily test in terms of RPS by limiting how many requests each VU is able to make per second.

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+Let's jump to example, and talk details later.
+
+{% highlight js linenos %}
+import { check, sleep } from "k6";
+import http from "k6/http";
+
+let desiredRPS = 200;
```
Comment (markjmeier) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: This is the total RPS for the test right?  We should comment that note.

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+$ k6 run -o cloud API_scenario_test.js
+{% endhighlight %}
+
+In both cases you will see a link to Load Impact where you can view your results in real time. It will look similar to this:
+
+![k6 output]({{ site.baseurl }}/assets/img/v4/how-to-tutorials/how-to-load-test-an-api/k6_cloud_output.png)
+
+
+
+Note, if you are just getting started or don't want to mess with a terminal, you can also execute scripts directly from Load Impact's web editor, without the need of installing k6. Go to the
```
Comment (markjmeier) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: "don't want to mess with a terminal" reads strange to me.  Maybe, this:  "...or don't want to execute tests from the command line"

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md (hunk 4 above)
Comment (markjmeier) [reply jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: The first sentence should also be reworded to be more direct/instructional/commanding.  "If you are testing one endpoint at a time, the RPS method is more meaningful".  This way we take all the ambiguity out of it. I guess you could test either the other way, but it is just more difficult to manage.

File: jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+  sleep(1); // user usually waits 1 second after this flow
+}
+{% endhighlight %}
+
+You can view a sample test run of this script [here](https://app.loadimpact.com/k6/anonymous/4eacae1dfd234566865962ccdb350c4e)
+
+![k6 rps with VUs]({{ site.baseurl }}/assets/img/v4/how-to-tutorials/how-to-load-test-an-api/rps_vus.png)
+
+Note, in this test run, the Request Rate closely follows VUs, while response time is relatively flat.
+This means that the API under test is handling traffic well for specified number or users.
```
Comment (markjmeier) [inline jekyll/_v4/how-to-tutorials/how-to-load-test-an-api.md]: Add another sentence so we reenforce a key concept:  Aside from some fringe cases, the aforementioned pattern is a sign of good performance.

**Pull request merged**